src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/DynamicDebuggerProxy.cs

```diff
@@ -0,0 +1,482 @@
+    internal sealed class DynamicBindingFailedException : Exception
+    {
+        public DynamicBindingFailedException()
+            : base()
+        {
+        }
+
+        private DynamicBindingFailedException(SerializationInfo info, StreamingContext context) 
+            : base(info, context) 
+        { 
+        }
+    }
+
+    internal sealed class GetMemberValueBinder : GetMemberBinder
+    {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode")]
+        public GetMemberValueBinder(string name, bool ignoreCase)
+            : base(name, ignoreCase)
+        {
+        }
+
+        public override DynamicMetaObject FallbackGetMember(DynamicMetaObject self, DynamicMetaObject onBindingError)
+        {
+            if (onBindingError == null)
+            {
+                var v = new List<DynamicMetaObject> { self };
+                var error = new DynamicMetaObject(System.Linq.Expressions.Expression.Throw(
+                    System.Linq.Expressions.Expression.Constant(new DynamicBindingFailedException(), typeof(Exception)), typeof(object)), System.Dynamic.BindingRestrictions.Combine(v));
+                return error;
+            }
+            return onBindingError;
+        }
+    }
+
+    internal sealed class DynamicMetaObjectProviderDebugView
+    {
+        [System.Diagnostics.DebuggerDisplay("{value}", Name = "{name, nq}", Type = "{type, nq}")]
+        internal class DynamicProperty
+        {
+            [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields")]
+            readonly string name;
+
+            [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields")]
+            readonly object value;
+
+            [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1823:AvoidUnusedPrivateFields")]
+            readonly string type;
+
+            [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1811:AvoidUncalledPrivateCode")]
+            public DynamicProperty(string name, object value)
+            {
+                this.name = name;
+                this.value = value;
+                this.type = value == null ? "<null>" : value.GetType().ToString();
+            }
+        }
+
+        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+        private IList<KeyValuePair<string, object>> results = null;
+
+        [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
+        private object obj;
```

Comment:
Since these aren't browsable, could they be _results and _obj?

Reply:
I'll leave these as is, in case the names are important.

---
In reply to: [295053099](https://github.com/dotnet/corefx/pull/38656#discussion_r295053099) [](ancestors = 295053099)